src/DynamoPackages/PackageLoader.cs

```diff
@@ -203,19 +203,23 @@ private void ScanAllPackageDirectories(IPreferences preferences)
 
         private void ScanPackageDirectories(string root, IPreferences preferences)
         {
-            if (!Directory.Exists(root))
+            try
             {
-                this.Log(string.Format(Resources.InvalidPackageFolderWarning, root));
-                return;
-            }
+                if (!Directory.Exists(root))
+                {
+                    this.Log(string.Format(Resources.InvalidPackageFolderWarning, root));
+                    return;
+                }
 
-            foreach (var dir in
-                Directory.EnumerateDirectories(root, "*", SearchOption.TopDirectoryOnly))
-            {
-                var pkg = ScanPackageDirectory(dir);
-                if (pkg != null && preferences.PackageDirectoriesToUninstall.Contains(dir)) 
-                    pkg.MarkForUninstall(preferences);
+                foreach (var dir in
+                    Directory.EnumerateDirectories(root, "*", SearchOption.TopDirectoryOnly))
+                {
+                    var pkg = ScanPackageDirectory(dir);
+                    if (pkg != null && preferences.PackageDirectoriesToUninstall.Contains(dir))
```

Comment:
Would `ScanPackageDirectory()` be an expensive operation? If it is, you may check `preferences.PackageDirectoriesToUninstall.Contains(dir)` before `ScanPackageDirectory()`